### PR TITLE
Add tests for SVG exports

### PR DESCRIPTION
**Pull request by IuryPiva**
- [x] relates to issue #11337
- [x] tests added / passed

Adds initial adaptations to the testing framework to handle debugging of the tests.
As states on #11337 more PRs will be open to cover the rest of the SVG backend

### CODE REVIEW

**Comment by IuryPiva** [comment]
@bryevdv running the `export_svg` function I noticed a drop in the quality of the final product, do you think this warrants a new issue?

With Bokeh plot:
image
Exported SVG:
image-2
With Bokeh plot:
image-3
Exported SVG:
image-4

**Comment by mattpap** [comment]
> running the export_svg function I noticed a drop in the quality of the final product, do you think this warrants a new issue?

It warrants having some place (discussion/issue), where you can actually provide a complete set of information for us to assess this. In particular, given that SVGs are involved, provide the actual files, so that we can preview them locally and establish, if this is a bokeh's or your system's issue. If it's a bokeh's issue, then it's very likely it's related to high DPI handling (do you have a retina or other type of display with `dpr != 1`?). What browser did `export_svg()` actually use?

**Comment by mattpap** [comment]
> Add tests for SVG exports

So far this PR adds an example. The only testing that happens here, is making sure that no exceptions or other run time failures don't occur on python and js side of things. The output is not compared in anyway. It's only stored in an artifact for manual inspection. Note there are already some basic unit tests for `get_svg()`, which is a basis for `export_svg()` and `export_svgs()`, so you may want to unit test those.

File: bokehjs/test/unit/core/util/svg.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+function compare_on_dom(fn: (ctx: CanvasRenderingContext2D) => void, svg: string, {width, height}: {width: number, height: number}) {
+  const canvas = document.createElement("canvas")
+  canvas.height = height
+  canvas.width = width
+
+  document.body.appendChild(canvas)
+
+  const ctx = canvas.getContext("2d") as CanvasRenderingContext2D
```
Comment (mattpap) [inline bokehjs/test/unit/core/util/svg.ts:18]: If this cast is all about dropping `null`, then use `!` postfix operator instead.

File: bokehjs/test/unit/core/util/svg.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    const test = (_ctx: SVGRenderingContext2D | CanvasRenderingContext2D) => {
+      // Draw shapes
+      for (let i = 0; i <= 3; i++) {
+        for (let j = 0; j <= 2; j++) {
+          _ctx.beginPath()
+          const x             = 25 + j * 50                 // x coordinate
+          const y             = 25 + i * 50                 // y coordinate
+          const radius        = 20                          // Arc radius
+          const startAngle    = 0                           // Starting point on circle
+          const endAngle      = Math.PI + (Math.PI * j) / 2 // End point on circle
```
Comment (mattpap) [inline bokehjs/test/unit/core/util/svg.ts:71]: I'm repeating myself, but this is not the naming convention that we use in bokehjs. This applies to the other PR as well.

File: bokehjs/test/unit/core/util/svg.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  })
+
+
+  it("should fill a red rectangle correctly", () => {
+    const size = {width: 50, height: 50}
+    const ctx = new SVGRenderingContext2D(size)
+
+    const test = (_ctx: SVGRenderingContext2D | CanvasRenderingContext2D) => {
```
Comment (mattpap) [inline bokehjs/test/unit/core/util/svg.ts:97]: Leading `_` is an indicator of an unused argument. Preferably rename this to `ctx`. Given that we use lexical scoping exclusively, then there is no room for confusion between both contexts. However, you could also shuffle this code around, as `size` and `ctx` are not used in `test`, so they could be defined at later, further clarifying the code.

Comment (mattpap) [reply bokehjs/test/unit/core/util/svg.ts:97]: As long as public API testing goes, `SVGRenderingContext2D` and `CanvasRenderingContext2D` should provide exactly (or almost exactly given my uncertainty about `drawImage()`). I'm guessing there may be a problem with svg context not implementing canvas context, though that should be fixed on its own. Currently svg context is defined as:
```ts
export class SVGRenderingContext2D /*implements CanvasRenderingContext2D*/ { ... }
```
but it should work with `implements` clause uncommented.

**Comment by bryevdv** [comment]
@IuryPiva just for the sake of consistency and comlpeteness, can you add a corresponding  `examples/howto/export_to_png.py` to go along with ` examples/howto/export_to_svg.py` ? It would be odd to have one but not the other.

**Comment by bryevdv** [comment]
Was there any discussion about the kind of tests? I think there are some places for literal SVG string tests but in general those seem like they might be a bit fragile. Was there any discussion of loading the output SVGs in a browser and doing some sort of visual test on the result?

**Comment by IuryPiva** [comment]
This is the first iteration on testing the SVG backend and as always open to feedback. 

I'm testing each part of the SVGContext differently, this first PR is only the simpler drawing methods and what I found worked the best was just comparing the outputs. In the transformation tests I checked the initial position of the paths after each command.

The output of the SVGs in the browser are just to let the developer know how it's looking comparing to the CanvasContext, it does not implement any real testing tool. More of a 'debugging' tool.

**Pull request merged**